test/functional/wallet_listtransactions.py

```diff
@@ -40,14 +40,14 @@ def run_test(self):
                             {"txid": txid},
                             {"category": "receive", "amount": Decimal("0.1"), "confirmations": 0})
         # mine a block, confirmations should change:
-        self.nodes[0].generate(1)
+        blockhash = self.nodes[0].generate(1)[0]
         self.sync_all()
         assert_array_result(self.nodes[0].listtransactions(),
                             {"txid": txid},
-                            {"category": "send", "amount": Decimal("-0.1"), "confirmations": 1})
+                            {"category": "send", "amount": Decimal("-0.1"), "confirmations": 1, "blockhash": blockhash, "blockheight": 202})
```

Comment:
In commit "rpc: Expose block height of wallet transactions" (36d904246aa574f65ecea52b546402c2712a1fc3)

In these python tests, I think it would be better to get heights programmatically calling `node.getblockheader` than to hardcode numeric heights. Would make this PR easier to review because it'd be easier to confirm expected height values are correct.Would also make the tests easier to maintain in the future if more code is added to generate blocks.

Reply:
Done, also replaced the other `getblock` with `getblockheader`.